modules/@apostrophecms/modal/ui/apos/components/AposModalShareDraft.vue

```diff
@@ -175,15 +175,24 @@ export default {
       });
     },
     generateShareUrl(aposShareKey) {
-      const url = new URL(`${location.origin}${this.doc._url}`);
+      const regex = /^https?:\/\//;
+      const docUrl = regex.test(this.doc._url)
+        ? this.doc._url
+        : `${location.origin}${this.doc._url}`;
 
-      const slug = apos.http.addQueryToUrl(this.doc._url, {
+      const url = new URL(docUrl);
+
+      const urlInfo = {
+        url: url.href
+      };
+
+      apos.bus.$emit('shared-draft-link', urlInfo);
```

Comment:
Is this emitted for overriding purposes?
Where can this be done in frontend?

Reply:
would it be better to override it on backend side?

Reply:
It was calculated on frontend side so that would be difficult. A project level developer will usually be able to do a simple string replacement here (if they need anything at all) that would not be general enough if it were a core feature.

Reply:
okay, thanks for the precision guys!